### ISSUE TITLE
Dropping links into the mpv window doesn't work.

### ISSUE THREAD

**Issue by Tungsten842** [mpv-player/mpv#9789]
### Important Information

Provide following Information:
- mpv 0.34.1
- nixos unstable
- nixos
- sway 1.7
- intel open source driver

### Reproduction steps
Select an url, drag it and drop it in to the mpv window that says: "Drop files or URLs
 to play here."
### Expected behavior
The video plays
### Actual behavior
Nothing happens
### Log file
[log.txt](https://github.com/mpv-player/mpv/files/7958100/log.txt)
### Sample files
Any video

**Comment by CounterPillow** [comment]
> Nothing shows up in the log.

post a `--log-file` file anyway

**Comment by QuantMint** [comment]
> > Nothing shows up in the log.
> 
> post a `--log-file` file anyway

Ok.

**Comment by ValZapod** [comment]
Тboth wayland and windows in admin (cmd.exe) are removeing this too. Very bothersome.

**Comment by Dudemanguy** [comment]
I assume you are using firefox. This actually works in chromium. The log is important though because it does show we actually get DND offer but it isn't actually read from the fd for some reason.

**Comment by Dudemanguy** [comment]
Somewhat fixed by #9821 but not totally. If you disable `focus_follows_mouse` in sway, then it will now work with gtk applications as the source of the drag and drop. However, this doesn't work with `focus_follows_mouse` on (the default setting). It's not clear to me if mpv, sway, or gtk is doing something wrong in this case. This works fine in weston though.

**Comment by WhyNotHugo** [comment]
[It's been reported to work on dolphin + kwin/wayland.](https://github.com/mpv-player/mpv/issues/10258#issuecomment-1145095134)

It doesn't work for me dragging from nemo to mpv on swaywm either.

[I posted some logs when running with `WAYLAND_DEBUG=1 mpv myvideo.mp4`](https://github.com/mpv-player/mpv/issues/10258#issue-1258453299):

It does seem like mpv gets the offer, accept/receives it, `finish`es it 150ms later, and then destroys it:

```
[3674982.890] wl_data_device@16.motion(175240324, 162.00000000, 61.00000000)
[3674982.953]  -> wl_data_offer@4278190081.accept(175240324, "text/uri-list")
[3675108.120] wl_data_device@16.drop()
[3675108.198]  -> wl_data_offer@4278190081.receive("text/uri-list", fd 20)
[3675108.248]  -> wl_data_offer@4278190081.finish()
[3675108.269] wl_data_device@16.leave()
[3675108.288] wl_keyboard@23.enter(210809, wl_surface@6, array[0])
[3675108.349] wl_keyboard@23.modifiers(210810, 0, 0, 16, 0)
[3675108.433] wl_data_device@16.data_offer(new id wl_data_offer@4278190080)
[3675108.468]  -> wl_data_offer@4278190081.destroy()
```

**Comment by WhyNotHugo** [comment]
Does `mpv` do anything different when handling drag-and-drop **if** it has focus? This seems to be the factor that makes it break.

**Comment by WhyNotHugo** [comment]
Can I log to stdout? It might be useful to align logs with `WAYLAND_DEBUG` output.

**Comment by WhyNotHugo** [comment]
Output of `MPV_VERBOSE=debug WAYLAND_DEBUG=1 mpv --log-file=/proc/self/fd/2 Tea_Consent.mp4 2> mpv.log`:

https://paste.sr.ht/~whynothugo/f2a227a566aa64ef23a6a00df0ebe9881471b09d

**Comment by Dudemanguy** [comment]
Hmm, I guess I didn't explain it in the issue. This is what is happening to you. When using `focus_follows_mouse` on sway, sway instantly sends another handle data offer event (separate from the original one) as soon as the surface receives focus. GTK SIGHUPs the pipe when this happens. This causes mpv to be unable to read the data and thus nothing happens. QT doesn't have this SIGHUP behavior which is why drags from say dolphin to mpv work. Another way to get this to work is to  *not* use `focus_follows_mouse`. If you turn that option off, sway doesn't send extra data offer events, so the pipe stays open and mpv reads it normally.

tl;dr this is a really weird intersection of behavior and I'm not really sure who is at fault here.

**Comment by WhyNotHugo** [comment]
> When using focus_follows_mouse on sway, sway instantly sends another handle data offer event (separate from the original one) as soon as the surface receives focus.

Isn't this data_offer maybe the clipboard...?

> GTK SIGHUPs the pipe when this happens.

What's the rationale for GTK doing this? It doesn't sound like it should.

Yeah, but turning off `focus_follows_mouse` makes windows focusing as pleasant as using windows 95. :sweat_smile:

**Comment by Dudemanguy** [comment]
I don't know the inner details. You can see the sway bug report linked up earlier if you want to look at some more wayland logs. Basically, the events sway sends with `focus_follows_mouse` is very different and causes GTK to behave differently (i.e. SIGHUP). It's been a while since I've looked at this, but I was never able to determine a time to read the pipe and guarantee that it data would actually be received (it's possible to read it too early).

**Comment by LaserEyess** [comment]
@WhyNotHugo https://github.com/swaywm/sway/issues/6906

**Comment by WhyNotHugo** [comment]
Indeed, sounds like it’s a gtk3 bug. Is there an issue for this downstream already, or should I open one myself?
